core/src/main/java/com/linecorp/armeria/client/endpoint/EndpointGroup.java

```diff
@@ -41,6 +46,50 @@ static EndpointGroup empty() {
         return StaticEndpointGroup.EMPTY;
     }
 
+    /**
+     * Returns an {@link EndpointGroup} that combines all the {@link Endpoint}s of {@code endpointGroups}.
+     * {@code endpointGroups} can be instances of {@link Endpoint} as well, any {@link EndpointGroup}s and
+     * {@link Endpoint} will all be combined into a single {@link EndpointGroup} that contains the total set.
+     */
+    static EndpointGroup of(EndpointGroup... endpointGroups) {
+        requireNonNull(endpointGroups, "endpointGroups");
+        return of(ImmutableList.copyOf(endpointGroups));
+    }
+
+    /**
+     * Returns an {@link EndpointGroup} that combines all the {@link Endpoint}s of {@code endpointGroups}.
+     * {@code endpointGroups} can be instances of {@link Endpoint} as well, any {@link EndpointGroup}s and
+     * {@link Endpoint} will all be combined into a single {@link EndpointGroup} that contains the total set.
+     */
+    static EndpointGroup of(Iterable<EndpointGroup> endpointGroups) {
+        requireNonNull(endpointGroups, "endpointGroups");
+
+        final List<EndpointGroup> groups = new ArrayList<>();
+        final List<Endpoint> staticEndpoints = new ArrayList<>();
+        for (EndpointGroup endpointGroup : endpointGroups) {
+            if (endpointGroup instanceof Endpoint) {
```

Comment:
Should we also spread the endpoints in the `StaticEndpointGroup`?

Reply:
Sure also realized I was missing tests for these special cases